console/console-init/ui/src/modules/iot-device/components/CreateMetadata/MetaDataRow.tsx

```diff
@@ -29,16 +29,43 @@ export const MetaDataRow: React.FC<IMetaDataRow> = ({
   rowIndex
 }) => {
   const metadataRow = metadataList[rowIndex];
-  // console.log("metadataList", metadataList, "type", metadataRow.type);
+  const [validationStatus, setValidationStatus] = useState<"default" | "error">(
+    "default"
+  );
 
   const onSelectType = (typeValue: string) => {
+    const validationStatus = getValidationStatus(typeValue, metadataRow.value);
+    setValidationStatus(validationStatus);
     let updatedTypeMetadata = [...metadataList];
     updatedTypeMetadata[rowIndex].type = typeValue;
     if (isObjectOrArray(typeValue as any))
       updatedTypeMetadata[rowIndex].value = "";
     setMetadataList(updatedTypeMetadata);
   };
 
+  const getValidationStatus = (type: string, value: string) => {
+    let validationStatus: "default" | "error" = "default";
+    switch (type) {
```

Comment:
convert type into lower case for safer side. 
switch(type?.toLowerCase())

Reply:
As this is not user input data, so there is no possibility for the case to change.

Reply:
I have also used enum here, so it will further eliminate such issues.